chat/src/hooks/useChatState.tsx

```diff
@@ -0,0 +1,227 @@
+import constate from "constate";
+import { useCallback, useEffect, useState } from "react";
+import {
+  ChatConversation,
+  ChatMessageProps,
+  ChatMessages,
+  ChatState,
+} from "../types/ChatTypes";
+import { v4 as uuidv4 } from "uuid";
+import Events from "../utils/events";
+import { EditorContext } from "./useEditorContext";
+import { sendRequestToExtension } from "./ExtensionCommunicationProvider";
+
+type ChatStateResponse = {
+  chatData: ChatState;
+  updateConversation: (conversation: ChatConversation) => void;
+  clearAllConversations: () => void;
+  currentConversation: ChatConversation | null;
+  conversationMessages: ChatMessages;
+  isBotTyping: boolean;
+  setIsBotTyping(isBotTyping: boolean): void;
+  addMessage(message: ChatMessageProps): void;
+  updateLastMessageWithEditorContext(editorContext: EditorContext): void;
+  submitUserMessage(userText: string): void;
+  setCurrentConversationData(conversation: ChatConversation): void;
+  createNewConversation(): void;
+};
+
+function useCreateChatState(): ChatStateResponse {
+  const [chatData, setChatData] = useState<ChatState>({
+    conversations: {},
+  });
+
+  const [
+    currentConversation,
+    setCurrentConversation,
+  ] = useState<ChatConversation | null>(null);
+
+  const [
+    conversationMessages,
+    setConversationMessages,
+  ] = useState<ChatMessages>([]);
+
+  const resetCurrentConversation = useCallback(() => {
+    setCurrentConversation(null);
+    setConversationMessages([]);
+  }, [setCurrentConversation, setConversationMessages]);
+
+  const clearMessages = useCallback(() => {
+    setConversationMessages([]);
+  }, [setConversationMessages]);
+
+  const setCurrentConversationData = useCallback(
+    (conversation: ChatConversation) => {
+      setCurrentConversation(conversation);
+      setConversationMessages(conversation.messages);
+    },
+    [setCurrentConversation, setConversationMessages]
+  );
```

Comment:
`useReducer` feel more appropriate here, but I understand it will be hard to refactor now

Reply:
I'll find some time to change this